Assets/Editor/ModelInspector.cs

```diff
@@ -36,43 +45,52 @@ internal class PreviewSettings
             public Material meshMultiPreviewMaterial;
             public Material wireMaterial;
             public Material lineMaterial;
-            
+
             public bool[] availableDisplayModes = Enumerable.Repeat(true, 7).ToArray();
             public bool[] availableUVChannels = Enumerable.Repeat(true, 8).ToArray();
+
+            public GameObject skinnedMeshGameObject;
```

Comment:
The actual game object seems to not be needed as a state variable. What you need is something like "Mesh bakedSkinnedMesh;". And it should not be in "settings" since it's not a setting, just baked mesh for preview.

- Rename CreateSkinnedMeshRenderer to BakeSkinnedMesh
- Have it only create the game object inside that function, bake it to the mesh, destroy the game object. No need to carry it around.
- Same for `m_BaseGameObjectForSkinnedMeshRenderer`, it's not needed.